tests/php/CampaignAdminTest.php

```diff
@@ -129,4 +134,72 @@ public function testFilters()
             $results
         );
     }
+
+    public function testReadCampaignResponse()
+    {
+        // Login as admin user
+        $group = $this->objFromFixture(Group::class, 'admins');
+        Permission::grant($group->ID, 'ADMIN');
+
+        $admin = $this->objFromFixture(Member::class, 'admin_user');
+        $this->logInAs($admin);
+
+        // Create new ChangeSet object
+        $changeset = ChangeSet::create();
+        $changeset->ID = '123';
+        $changeset->Name = 'changeset 123';
+        $changeset->State = 'open';
+        $changeset->IsInferred = false;
+        
+        $changeset->write();
+
+        // ChangeSet should be accessable for Admin
+        $this->assertTrue($changeset->canView());
+
+        $expectedStatus = [
+            '200' => [
+                'ID' => '123',
+                'Name' => 'show'
+            ],
+            '400' => [
+                'ID' => '123',
+                'Name' => ''
+            ],
+            '404' => [
+                'ID' => '12345',
+                'Name' => 'show'
+            ]
+        ];
+
+        // Create new CampaignAdmin object
+        $campaignAdmin = CampaignAdmin::create();
+
+        $request = new HTTPRequest('GET', '/admin/campaigns/set/');
```

Comment:
There's a better way of doing this. If you change the parent class of this test to `FunctionalTest`, you can do calls like `$this->get("/admin/campaigns/set/$changeSetID/show", null, ['Accept' => 'application/json']);`

Reply:
DONE